templates/common/vsphere/units/nodeip-configuration.service.yaml

```diff
@@ -0,0 +1,46 @@
+name: nodeip-configuration.service
+enabled: true
+contents: |
+  [Unit]
+  Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+  # This is more effective in VIP managing environments where the kubelet and crio IP
+  # address picking logic is flawed and may end up selecting an address from a
+  # different subnet or a deprecated address
+  Wants=network-online.target crio-wipe.service
+  After=network-online.target ignition-firstboot-complete.service crio-wipe.service
+  Before=kubelet.service crio.service
+
+  [Service]
+  # Need oneshot to delay kubelet
+  Type=oneshot
+  # Would prefer to do Restart=on-failure instead of this bash retry loop, but
+  # the version of systemd we have right now doesn't support it. It should be
+  # available in systemd v244 and higher.
+  ExecStart=/bin/bash -c " \
+    until \
+    /usr/bin/podman run --rm \
+    --authfile /var/lib/kubelet/config.json \
+    --net=host \
+    --volume /etc/systemd/system:/etc/systemd/system:z \
+    {{ .Images.baremetalRuntimeCfgImage }} \
+    node-ip \
+    set --retry-on-failure \
+    {{if eq .IPFamilies "IPv6" -}}
+    --prefer-ipv6 \
+    {{end -}}
+    {{if (onPremPlatformAPIServerInternalIP .) -}}
```

Comment:
out of curiosity, why the extra if here? if it was empty the next line would just be `;\` like the else condition I think

Reply:
That's what I thought too! But the templating engine actually printed it as `<no value>` which then broke the unit, so I had to add this `if` in to prevent it trying to print the empty value. (This took me ages to work out 😓 )